modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -777,6 +779,14 @@ CV_EXPORTS_W int solveP3P( InputArray objectPoints, InputArray imagePoints,
                            OutputArrayOfArrays rvecs, OutputArrayOfArrays tvecs,
                            int flags );
 
+//TODO: doc
+CV_EXPORTS_W int solvePnPGeneric( InputArray objectPoints, InputArray imagePoints,
```

Comment:
Other possibility could be to return a `struct` of pose info instead of all the parameters. But I have not seen this yet in OpenCV, due to issue with Python/Java bindings maybe?

Reply:
> Python/Java bindings

I believe, bindings generator should support non-complex structures.